### PR TITLE
BUG FIX: Enable remote access to DB server

### PR DESCRIPTION
**Pull request by paolopaolopaolo**
- Add `pg_hba_config.j2` template
- Add module calls to add postgres config file in `db/tasks/main.yml`
- Add `db_host` variable to env_vars
- Utilize `private_application_host` and `app_host_ip` correctly

@rmutter, this re-enables having application hosts hit database hosts when they aren't on the same server. The current weakness of the system is that if you want to add more hosts running applications, those host numbers need to be added manually in the appropriate `orchestration/env_vars` file (specifically in the `private_application_hosts` array).

### CODE REVIEW

File: yurt/yurt_core/tests/add_test.py [hunk 1]
```diff
@@ -38,8 +38,9 @@ def test_remote_server(self,
             'git_repo': 'git@github.com:paolopaolopaolo/the-monster-mash.git',
             'env': 'itWasAGraveyardSmash',
             'abbrev_env': 'theyDidTheMash',
-            'app_host_ip': 'it-caught-on-in-a-flash.com',
-            'db_host_ip': 'he-did-the-mash.org',
+            'app_host_dns': 'it-caught-on-in-a-flash.com',
+            'app_host_ip': '1.2.3.4',
```
Comment (rmutter) [inline yurt/yurt_core/tests/add_test.py]: had to ruin your monster mash joke 😭

**Comment by rmutter** [comment]
👍

File: yurt/orchestration/roles/db/tasks/main.yml [hunk 2]
```diff
@@ -12,24 +12,44 @@
...
+  become_user: postgres
   postgresql_db: name={{ db_name }}
                  encoding='UTF-8'
                  lc_collate='en_US.UTF-8'
                  lc_ctype='en_US.UTF-8'
                  template='template0'
                  state=present
 
+- name: Ensure postgres is listening for application servers
+  lineinfile: dest=/etc/postgresql/9.4/main/postgresql.conf
```
Comment (rmutter) [inline yurt/orchestration/roles/db/tasks/main.yml]: `9.4` could change in the future, not sure what's the best strategy here

File: yurt/orchestration/roles/db/tasks/main.yml [hunk 3]
```diff
@@ -12,24 +12,44 @@
...
-  sudo_user: postgres
+  become_user: postgres
   postgresql_user: name={{ db_user }}
                    role_attr_flags=NOSUPERUSER,NOCREATEDB
-                   state=present
\ No newline at end of file
+                   state=present
+
+- name: Ensure application servers can access the db server
+  template: src=pg_hba_config.j2
```
Comment (rmutter) [inline yurt/orchestration/roles/db/tasks/main.yml]: did you forget to `git add pg_hba_config.j2`?

File: yurt/orchestration/roles/db/tasks/main.yml (hunk 2 above)
Comment (paolopaolopaolo) [inline yurt/orchestration/roles/db/tasks/main.yml]: Made the postgresql installations static in the latest commit

File: yurt/orchestration/roles/db/tasks/main.yml (hunk 3 above)
Comment (paolopaolopaolo) [inline yurt/orchestration/roles/db/tasks/main.yml]: Added (had a messed up `.gitignore`)!

**Comment by paolopaolopaolo** [comment]
Made a commit that fixes some stuff. Is it a good idea to make the installation for postgresql static like I have here?

**Comment by rmutter** [comment]
@paolopaolopaolo is it possible to keep it pinned to major versions like postgres 9.4 but be able to get the latest minor version, i.e. 9.4.3 vs. 9.4.7 so that it includes the latest bug fixes for that version? For PostgreSQL and likewise also python-psycopg I like the idea of keep yurt currently being 9.4 compatible and it needing to be upgraded on purpose when we know it supports 9.5.

**Comment by paolopaolopaolo** [comment]
@rmutter, totally possible (just set to `9.4.*`). Tested on a personal project and it installed just fine. Good to merge?

**Comment by rmutter** [comment]
@paolopaolopaolo awesome 👍 - merge away!

**Pull request merged**